docs/PythonAPI.md

```diff
@@ -0,0 +1,867 @@
+
+```python
+class AvailableData:
+    def frames(self) -> Iterator[VideoFrame]:
+        """Returns an iterator over the video frames in the available data."""
+    
+    def get_frame_count(self) -> int:
+        """Returns the total number of video frames in the available data."""
+    
+    def __iter__(self) -> Iterator[VideoFrame]:
+        """Returns an iterator over the video frames in the available data."""
+```
+- Call `get_frame_count()` to query the number of frames in an `AvailableData` object. 
+
+- The `frames` method provides an iterator over these frames.
+
+## Class `Camera`
+The `Camera` class is used to describe cameras or other video sources.
+
+```python
+class Camera:
+    identifier: Optional[DeviceIdentifier]
+    settings: CameraProperties
+
+    def __init__(self, *args: None, **kwargs: Any) -> None: ...
+    """Initializes a Camera object with optional arguments."""
+    
+    def dict(self) -> Dict[str, Any]: ...
+    """Returns a dictionary of the Camera attributes."""
+```
+
+- `identifier`: An optional attribute which contains an instance of the `DeviceIdentifier` class that describes the camera, or video source, if that device is natively supported. Otherwise, it is of type `None`.
+
+- `settings`: An instance of the `CameraProperties` class which contains the settings for the camera.
+
+- The `dict` method creates a dictionary of a `Camera` object's attributes.
+
+## Class `CameraProperties`
+The `CameraProperties` class is used to set the desired camera properties for acquisition.
+
+```python
+class CameraProperties:
+    exposure_time_us: float
+    line_interval_us: float
+    binning: float
+    pixel_type: SampleType
+    readout_direction: Direction
+    offset: Tuple[int, int]
+    shape: Tuple[int, int]
+    input_triggers: InputTriggers
+    output_triggers: OutputTriggers
+
+    def __init__(self, *args: None, **kwargs: Any) -> None: ...
+    """Initializes a CameraProperties object with optional arguments."""
+    
+    def dict(self) -> Dict[str, Any]: ...
+    """Returns a dictionary of the CameraProperties attributes."""
+```
+
+- `exposure_time_us`: How long in microseconds your camera should collect light from the sample. However, for simulated cameras, this is just a waiting period before generating the next frame.
+
+- `line_interval_us`: The time to scan one line in microseconds in a rolling shutter camera camera.
+
+- `binning`: How many adjacent pixels in each direction to combine by averaging. If binning is set to 1, no pixels will be combined.
+
+- `pixel_type`: An instance of the `SampleType` class which specifies the numerical data type.
+
+- `readout_direction`: An instance of the `Direction` class which specifies whether the data is readout forwards or backwards.
+
+- `offset`: A tuple of two integers representing the dark counts of the camera.
+
+- `shape`: A tuple of two integers representing the shape of the camera detector in pixels.
```

Comment:
What do the values in the tuple represent?

Reply:
Like offset, I think this describes the shape of the image region of interest. Again, the values represent pixels after binning for spinnaker/genicam.